ui-src/lib/views/CalendarView.js

```diff
@@ -0,0 +1,388 @@
+    this._otherDayElement.prop('checked', !this._sameDay);
+    this._availableDates = null;
+    this._minDateStr = null;
+    this._maxDateStr = null;
+    this._singleDateStr = null;
+    this._cameraViews = null;
+    this._rangeChangedHandler = null;
+  }
+
+  /**
+   * Change timezone.
+   *
+   * @param  {String} tz New timezone
+   */
+  set tz(tz) {
+    this._timeParser.tz = tz;
+  }
+
+  /**
+   * (Re)configure the datepickers and other calendar range inputs to reflect
+   * available dates.
+   */
+  _configureDatePickers() {
+    const dateSet = this._availableDates;
+    const minDateStr = this._minDateStr;
+    const maxDateStr = this._maxDateStr;
+    const fromPickerView = this._fromPickerView;
+    const toPickerView = this._toPickerView;
+    const beforeShowDay = function(date) {
+      let dateStr = date.toISOString().substr(0, 10);
+      return [dateSet.has(dateStr), '', ''];
+    };
+
+    if (this._sameDay) {
+      fromPickerView.option({
+        dateFormat: $.datepicker.ISO_8601,
+        minDate: minDateStr,
+        maxDate: maxDateStr,
+        onSelect: (dateStr, picker) => this._updateRangeDates(dateStr, dateStr),
+        beforeShowDay: beforeShowDay,
+        disabled: false,
+      });
+      toPickerView.destroy();
+      toPickerView.activate(); // Default state, but alive
+    } else {
+      fromPickerView.option({
+        dateFormat: $.datepicker.ISO_8601,
+        minDate: minDateStr,
+        onSelect: (dateStr, picker) => {
+          toPickerView.option('minDate', this.fromDateISO);
+          this._updateRangeDates(dateStr, this.toDateISO);
+        },
+        beforeShowDay: beforeShowDay,
+        disabled: false,
+      });
+      toPickerView.option({
+        dateFormat: $.datepicker.ISO_8601,
+        minDate: fromPickerView.dateISO,
+        maxDate: maxDateStr,
+        onSelect: (dateStr, picker) => {
+          fromPickerView.option('maxDate', this.toDateISO);
+          this._updateRangeDates(this.fromDateISO, dateStr);
+        },
+        beforeShowDay: beforeShowDay,
+        disabled: false,
+      });
+      toPickerView.date = fromPickerView.date;
+      fromPickerView.maxDate = toPickerView.date;
+    }
+  }
+
+  /**
+   * Call an installed handler (if any) to inform that range has changed.
+   */
+  _informRangeChange() {
+    if (this._rangeChangedHandler !== null) {
+      this._rangeChangedHandler(this._selectedRange);
+    }
+  }
+
+  /**
+   * Handle a change in the time input of either from or to.
+   *
+   * The change requires updating the selected range and then informing
+   * any listeners that the range has changed (so they can update data).
+   *
+   * @param  {String}  newTimeStr Time string from input element
+   * @param  {Boolean} isEnd      True if this is for end time
+   */
+  _pickerTimeChanged(newTimeStr, isEnd) {
+    console.log('newTimeStr: ' + newTimeStr);
+    const selectedRange = this._selectedRange;
+    parsedTS = isEnd
```

Comment:
Shouldn't there be a "let" or "const" here? I get an error when I set a start time:

```
Uncaught ReferenceError: parsedTS is not defined
    at CalendarView._pickerTimeChanged (CalendarView.js:208)
    at handler (CalendarView.js:299)
    at HTMLInputElement._startTimeElement.change (CalendarView.js:301)
    at HTMLInputElement.dispatch (jquery.js:5206)
    at HTMLInputElement.elemData.handle (jquery.js:5014)
```

Reply:
Oops. I have never heretofore used the time input, so forgot to test it. It needed a const, but that did not quite fix things, so I fixed the other two small issues coming after as well.